### PR TITLE
[Layout foundations] Update layout component docs and guidance

### LINKED ISSUE #8315: [Layout foundations] [Beta] Update docs/guidance for layout primitives in style guide
[Figma](https://www.figma.com/file/5oEi1wm73KhGLN0VS02oYm/Layout-components?node-id=3304%3A50565)

### PR DESCRIPTION
**Pull request by nayeobkim**
### WHY are these changes introduced?

Resolves #8315.

Updates layout component documentation and guidance in preparation for marking them beta in v11.

### WHAT is this pull request doing?

Updates `AlphaCard`, `AlphaStack`, `Inline`, `Bleed`, `Columns`, and `Box` documentation.

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by aveline** [comment]
Using the new color tokens is blocked by https://github.com/Shopify/polaris/pull/8414. We should wait until that is merged in and make any necessary updates to custom property names before merging in this PR.

File: polaris.shopify.com/content/components/layout-and-structure/alpha-stack.md [hunk 1]
```diff
@@ -1,29 +1,37 @@
...
   - layout
+  - stack
+  - responsive
+  - rows
+  - vertical centering
+  - horizontal row of components
+  - flexible items
+  - flex
+  - flexbox
+  - row of components
```
Comment (kyledurand) [inline polaris.shopify.com/content/components/layout-and-structure/alpha-stack.md]: 🔥

File: polaris.shopify.com/content/components/layout-and-structure/alpha-stack.md [hunk 2]
```diff
@@ -1,29 +1,37 @@
 ---
 title: Alpha stack
-description: Use to display elements vertically with items placed directly below the preceding item.
+description: Use to display children vertically with full width by default. Based on CSS Flexbox.
 category: Layout and structure
 keywords:
   - layout
+  - stack
+  - responsive
+  - rows
```
Comment (kyledurand) [inline polaris.shopify.com/content/components/layout-and-structure/alpha-stack.md]: 🔥

File: polaris.shopify.com/content/components/layout-and-structure/alpha-stack.md [hunk 3]
```diff
@@ -1,29 +1,37 @@
...
+  - vertical centering
+  - horizontal row of components
```
Comment (kyledurand) [inline polaris.shopify.com/content/components/layout-and-structure/alpha-stack.md]: 🔥

File: polaris.shopify.com/content/components/layout-and-structure/inline.md [hunk 4]
```diff
@@ -1,35 +1,36 @@
...
 title: Inline
-description: Use to arrange items in a horizontal row with equal spacing around them. Items wrap onto multiple lines when needed.
+description: Use to display children horizontally in a row. Based on CSS Flexbox.
 category: Layout and structure
 keywords:
   - layout
+  - inline
+  - flexbox
+  - flex
+  - responsive
```
Comment (kyledurand) [inline polaris.shopify.com/content/components/layout-and-structure/inline.md:10]: Can we add the 🔥 ones from Stack here?

**Review by laurkim — APPROVED** [review]

**Review by kyledurand — APPROVED** [review]

**Pull request merged**